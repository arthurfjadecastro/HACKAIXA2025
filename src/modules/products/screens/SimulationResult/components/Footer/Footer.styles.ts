import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  footer: {
    paddingHorizontal: 24,
    paddingVertical: 16,
    backgroundColor: '#F4F8FB',
  },

  newSimulationButton: {
    backgroundColor: '#F7931E',
    borderRadius: 8,
    height: 56,
    justifyContent: 'center',
  },
});
