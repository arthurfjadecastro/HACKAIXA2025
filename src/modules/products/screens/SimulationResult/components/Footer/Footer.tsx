import React from 'react';
import { View } from 'react-native';
import { Button } from '@/design-system/components';
import { styles } from './Footer.styles';

interface FooterProps {
  onNewSimulation: () => void;
}

export const Footer: React.FC<FooterProps> = ({ onNewSimulation }) => {
  return (
    <View style={styles.footer}>
      <Button
        title="Nova Simulação"
        onPress={onNewSimulation}
        style={styles.newSimulationButton}
      />
    </View>
  );
};
