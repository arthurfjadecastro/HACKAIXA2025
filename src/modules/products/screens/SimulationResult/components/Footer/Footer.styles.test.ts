import { styles } from './Footer.styles';

describe('Footer.styles', () => {
  it('should export styles object', () => {
    expect(styles).toBeDefined();
    expect(typeof styles).toBe('object');
  });

  it('should have footer style', () => {
    expect(styles.footer).toBeDefined();
    expect(styles.footer.paddingHorizontal).toBe(24);
    expect(styles.footer.paddingVertical).toBe(16);
  });

  it('should have newSimulationButton style', () => {
    expect(styles.newSimulationButton).toBeDefined();
    expect(styles.newSimulationButton.backgroundColor).toBe('#F7931E');
    expect(styles.newSimulationButton.borderRadius).toBe(8);
  });

  it('should have consistent style structure', () => {
    expect(styles).toMatchSnapshot();
  });
});
