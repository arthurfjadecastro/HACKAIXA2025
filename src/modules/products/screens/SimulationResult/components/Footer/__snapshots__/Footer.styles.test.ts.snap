// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.styles should have consistent style structure 1`] = `
{
  "footer": {
    "backgroundColor": "#F4F8FB",
    "paddingHorizontal": 24,
    "paddingVertical": 16,
  },
  "newSimulationButton": {
    "backgroundColor": "#F7931E",
    "borderRadius": 8,
    "height": 56,
    "justifyContent": "center",
  },
}
`;
